src/convertToObject.js

```diff
@@ -6,7 +6,36 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (sourceString.length === 0) {
+    return {};
+  }
+
+  const newStyles = sourceString.trim();
+  const stylesArray = newStyles.split(';');
+  const arr = [];
+
+  for (let i = 0; i < stylesArray.length; i++) {
+    if (stylesArray[i].trim().length !== 0) {
+      arr.push(stylesArray[i].replace('\n', ''));
+    }
```

Comment:
[CODE KNOWLEDGE] - Don't need to make copy before each operation. Moreover, methods like `filter` or `map` don't mutate original array. Instead of creating an intermediate array 'arr' and using a for loop, you could chain `split`, `filter`, and `map` directly on the result of `sourceString.trim().split(';')`.